### ISSUE TITLE
`vi.unmock` does not unmock imitated module

### ISSUE THREAD

**Issue by mcous** [vitest-dev/vitest#2351]
### Describe the bug

When mocking modules using `vi.mock` (or `vi.doMock`), if the module is replaced implicitly, a subsequent call to `vi.unmock` (or `vi.doUnmock`) will _not_ unmock the module.

```typescript
import {vi, afterEach, test, expect} from 'vitest'

afterEach(() => {
  vi.doUnmock('./dependency')
})

test('mock by imitating the actual module', async () => {
  vi.doMock('./dependency')
  const dependency = await import('./dependency')
  expect(vi.mocked(dependency.doSomething).mock).toBeDefined()
})

test('verify module is not mocked after call to doUnmock', async () => {
  const dependency = await import('./dependency')
  expect(vi.mocked(dependency.doSomething).mock).not.toBeDefined()
})
```

I believe this to be a bug, because if you provide an _explicit_ replacement via the second argument to `vi.mock`, instead of relying on the module imitation logic, the call to `unmock` will successfully remove the mock.

### Reproduction

See [mcous/vitest-unmock-bug](https://github.com/mcous/vitest-unmock-bug)

### System Info

```shell
System:
    OS: macOS 12.5.1
    CPU: (8) x64 Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz
    Memory: 2.06 GB / 16.00 GB
    Shell: 5.9 - /usr/local/bin/zsh
  Binaries:
    Node: 16.18.1 - ~/.nvs/default/bin/node
    npm: 8.19.2 - ~/.nvs/default/bin/npm
    Watchman: 2022.11.07.00 - /usr/local/bin/watchman
  Browsers:
    Firefox: 107.0
    Safari: 15.6.1
  npmPackages:
    vite: ^3.2.4 => 3.2.4
    vitest: ^0.25.2 => 0.25.2
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sheremet-va** [comment]
This should have „in“ check, because mocks without factory are stored as „null“:

https://github.com/vitest-dev/vitest/blob/24998285595e9f35b72dcf7a32880698d803dda4/packages/vitest/src/runtime/mocker.ts#L294

PR welcome

**Comment by sheremet-va** [comment]
Fixed by #2353
